File: Simpsons.R
library(tidyverse)
library(ggthemes)
library(patchwork)
setwd("C:/Users/meala/Documents/R Data Work")

#AreaSD
set.seed(259)
AreaSD_hours <- rnorm(100, mean = 90, sd = 15)
AreaSD_Income <-  abs(AreaSD_hours * (rnorm(100, 120,10))) 
SD <- ggplot(data.frame(AreaSD_hours, AreaSD_Income), aes(x = AreaSD_hours, y = AreaSD_Income)) + 
                  geom_point(color="darkgreen")  + labs(title = "Area SD", x = "Hours", y = "Income")+theme_classic()

#AreaLA
set.seed(259)
AreaLA_hours <- rnorm(90, mean = 100, sd = 16)
AreaLA_Income <-  abs(AreaLA_hours * (rnorm(90, 100,8)))
LA<- ggplot(data.frame(AreaLA_hours, AreaLA_Income), aes(x = AreaLA_hours, y = AreaLA_Income)) + 
                  geom_point(color="darkblue")  + labs(title = "Area LA", x = "Hours", y = "Income")+theme_classic()


#AreaB
set.seed(259)
AreaB_hours <- rnorm(110, mean = 120, sd = 20)
AreaB_Income <- abs(AreaB_hours * (rnorm(110, 75,6)))
Berkeley <- ggplot(data.frame(AreaB_hours, AreaB_Income), aes(x = AreaB_hours, y = AreaB_Income)) + 
                  geom_point(color="darkorange")  + labs(title = "Area SF", x = "Hours", y = "Income")+theme_classic()


#Irvine
set.seed(259)
AreaI_hours <- rnorm(80, mean = 170, sd = 16)
AreaI_Income <-  abs(AreaI_hours * (rnorm(80, 50,5)))

Irvine <- ggplot(data.frame(AreaI_hours, AreaI_Income), aes(x = AreaI_hours, y = AreaI_Income)) + 
                  geom_point(color="darkred")  + labs(title = "Area Irvine", x = "Hours", y = "Income")+theme_classic()

#Riverside
set.seed(259)
AreaR_hours <- rnorm(75, mean = 190, sd = 14)
AreaR_Income <-  abs(AreaR_hours * (rnorm(65, 45,4)))
Riverside <- ggplot(data.frame(AreaR_hours, AreaR_Income), aes(x = AreaR_hours, y = AreaR_Income)) + 
                  geom_point(color="brown")  + labs(title = "Area Riverside", x = "Hours", y = "Income")+theme_classic()

#rm(list=ls())
#plot all the plots together
SD
LA
Berkeley
Irvine
Riverside

SD+LA + Berkeley + Irvine + Riverside + plot_layout(ncol = 2)

#make a function to run linear regression 

AreaSD <- lm(AreaSD_Income~AreaSD_hours)
summary(AreaSD)
AreaLA <- lm(AreaLA_Income~AreaLA_hours)
summary(AreaLA)
AreaB <- lm(AreaB_Income~AreaB_hours)
summary(AreaB)
AreaI <- lm(AreaI_Income~AreaI_hours)
summary(AreaI)
AreaR <- lm(AreaR_Income~AreaR_hours)
summary(AreaR)

AreaSD <- data.frame(Area = "AreaSD",Hours= AreaSD_hours,Income=AreaSD_Income)
AreaLA <- data.frame(Area = "AreaLA",Hours= AreaLA_hours,Income=AreaLA_Income)
AreaB <- data.frame(Area = "AreaB",Hours= AreaB_hours,Income=AreaB_Income)
AreaI <- data.frame(Area = "AreaI",Hours= AreaI_hours,Income=AreaI_Income)
AreaR <- data.frame(Area = "AreaR",Hours= AreaR_hours,Income=AreaR_Income)

All <- rbind(AreaSD,AreaLA,AreaB,AreaI,AreaR)
summary(lm(data=All, Income~Hours))

p <- ggplot(data=All, aes(x=Hours, y=Income)) + geom_point()+geom_point(color="pink")
p<- p + theme_classic()

p1 <- ggplot(data=All, aes(x=Hours, y=Income, color=Area)) + geom_point()+geom_smooth(method = "lm", se = FALSE, color = "red")
p1<- p1 + theme_classic()

p2 <- ggplot(data=All, aes(x=Hours, y=Income, color=Area)) + geom_point()+geom_smooth(method = "lm", se = FALSE)
p2<- p2 + theme_classic()

p+p1+p2

Area_reg <- lm(data=All, Income~Hours+Area)
summary(Area_reg)

File: Simpsons.py
#!/usr/bin/env python
# coding: utf-8

# In[10]:


import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.api as sm

# Setting the seed for reproducibility
np.random.seed(259)

# Function to generate data for each area
def generate_data(mean_hours, sd_hours, mean_income, sd_income, n, base_income, area_name):
    hours = np.random.normal(mean_hours, sd_hours, n)
    income = 0.25 * np.abs(hours * np.random.normal(mean_income, sd_income, n)) + base_income
    return pd.DataFrame({'Hours': hours, 'Income': income, 'Area': area_name})

# Generate data for each area
AreaSD_df = generate_data(90, 3, 105, 5, 100, 700, 'AreaSD')
AreaLA_df = generate_data(92, 3, 97, 6, 90, 500, 'AreaLA')
AreaB_df = generate_data(93, 3, 90, 6, 110, 100, 'AreaB')
AreaI_df = generate_data(93, 4, 73, 5, 80, 0, 'AreaI')
AreaR_df = generate_data(96, 3, 68, 4, 75, -300, 'AreaR')

# Combine all areas into a single DataFrame
all_data = pd.concat([AreaSD_df, AreaLA_df, AreaB_df, AreaI_df, AreaR_df])


# In[11]:


# Plotting for each area
def plot_area(data, color, title):
    sns.scatterplot(x='Hours', y='Income', data=data, color=color)
    plt.title(title)
    plt.xlabel("Hours")
    plt.ylabel("Income")
    plt.show()

plot_area(AreaSD_df, "darkgreen", "Area SD")
plot_area(AreaLA_df, "darkblue", "Area LA")
plot_area(AreaB_df, "darkorange", "Area Berkeley")
plot_area(AreaI_df, "darkred", "Area Irvine")
plot_area(AreaR_df, "brown", "Area Riverside")


# In[12]:


# Perform linear regression for each area
def perform_regression(x, y):
    x_with_const = sm.add_constant(x)
    model = sm.OLS(y, x_with_const).fit()
    print(model.summary())

for df in [AreaSD_df, AreaLA_df, AreaB_df, AreaI_df, AreaR_df]:
    perform_regression(df['Hours'], df['Income'])

# Perform linear regression on the combined data
all_data_encoded = pd.get_dummies(all_data, columns=['Area'], drop_first=True)
model_all = sm.OLS(all_data_encoded['Income'], sm.add_constant(all_data_encoded.drop('Income', axis=1))).fit()
print(model_all.summary())

# Plotting combined data with regression line
sns.lmplot(x="Hours", y="Income", hue="Area", data=all_data, ci=None)
plt.show()

